Vincent http server ip

Vincentmask opened the pull request:
fixed UI tests
fixed server can't get/post across different machine
changed IP address to get the server working across machines

The website needs to fetch a special IP address to interact with the server, couldn't work in previous code when fetching localhost.

Jeff fixed the background color
Moved the title 9 subtitle above the submit button

Vincentmask committed:
added xss package

Vincentmask committed:
sanitize input

Vincentmask committed:
fix logo image missing issue

Vincentmask committed:
we don't need this for now

Vincentmask committed:
all tests pass

jeffmchale committed:
Background color fix

jeffmchale committed:
Moved Title 9 subtitle above button

Vincentmask committed:
added ip address to server for http for other machine

Vincentmask committed:
changed the ip address

imOnVacation reviewed (approved):
Thanks, Jeff and Vincent.

Vincent, can you explain why we are adding the IP address to the server file?

MrFiszbi1 commented on src/safeUI/SafeUI.tsx:
```diff
@@ -121,7 +121,7 @@ function SafeUI() {
       // fetch data from API endpoint
       // param: request method, header, body
-      fetch('http://localhost:3001/addMessage', {
+      fetch('http://131.252.208.28:3001/addMessage', {
```
I am not totally sure, but I am not sure if this is safe. In the full stack class, I am taking we put the host and port IP addresses in a separate .env file. Maybe I am misunderstanding the two things. Not sure, do you know what I am talking about?

NAlexH2 replied:
I'm with Jafar on this. This enables various attack vectors.

Vincentmask replied:
Yeah that was my concert at the beginning as well. I am thinking to have this as part of the script like the database info we had

Vincentmask committed:
roll back to localhost, we can hide the ip by using
the safe domain after the VM is set.
Now the code can only work on one machine
that is running the server

Vincentmask commented:
Change the port back to localhost. Once we got the VM and domain working we can hide the IP address later

Vincentmask committed:
fix the title 9 back to where it belongs,
looks like I messed it up last commit
